en/docs/install-and-setup/setup/security/logins-and-passwords/harshicrop-vault-extension.md

```diff
@@ -0,0 +1,298 @@
+        ```
+
+    !!! note
+        When you add "tmp" to the file name, note that this will automatically get deleted from the file system after the server starts. Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
+
+        === "Linux"
+            ```
+            The file name should be `hashicorpRootToken-persist`.
+            ```
+        
+        === "Windows"
+            ```
+            The file name should be `hashicorpRootToken-persist.txt`.
+            ```
+
+
+2. Start the WSO2 API Manager Server and enter the keystore password at startup when prompted:
+
+    ```bash
+    [Enter KeyStore and Private Key Password :] wso2carbon
+    ```
+
+---
+
+## Setting up using AppRole authentication
+
+### Step 1: Setup HashiCorp Vault
+
+1. Start the HashiCorp Vault server and set the environment variables:
+
+    ```bash
+    export VAULT_ADDR='http://127.0.0.1:8200'
+    export VAULT_TOKEN='<root token>'
+    ```
+
+2. Include the policy in the `kv-read-write.hcl` file as below:
+
+    ```hcl
+    path "wso2apim/data/*" {
+        capabilities = ["create", "read", "update", "delete", "list"]
+    }
+    ```
+
+3. Upload the kv policy as below:
+
+    ```bash
+    vault policy write kv-read-write kv-read-write.hcl
+    ```
+
+4. The following commands can be used to create the AppRole using the created policy:
+
+    ```bash
+    vault auth enable approle
+    ```
+    ```bash   
+    vault write auth/approle/role/my-role \
+        token_policies="kv-read-write" \
+        token_type="service" \
+        token_ttl="24h" \
+        token_max_ttl="72h"
+    ```
+
+5. Get the role-id as below:
+
+    ```bash
+    vault read auth/approle/role/my-role/role-id
+    ```
+
+6. Get the secret-id as below:
+
+    ```bash
+    vault write -f auth/approle/role/my-role/secret-id
+    ```
+
+7. Create the kv engine with a path name (e.g., `wso2apim`).
+
+    The following commands can be used to add secrets with the HashiCorp Vault:
+
+    ```bash
+    # Create a new kv engine
+    vault secrets enable -path=wso2apim -version=2 kv
+
+    # Add new secret
+    vault kv put wso2apim/admin_password value=admin
+
+    # Get a secret (To check)
+    vault kv get -field=value wso2apim/admin_password
+    ```
+
+### Step 2: Configure HashiCorp Vault extension
+
+1. Clone [HashiCorp Vault extension](https://github.com/wso2-extensions/carbon-securevault-hashicorp/tree/master) repo
+
+2. Build the HashiCorp Vault Integration OSGI bundle using `mvn clean install` and copy the `target/org.wso2.carbon.securevault.hashicorp-1.0.jar` file to the `<APIM_HOME>/repository/components/dropin/`
+   directory.
+
+3. Add **HashiCorp Vault Java Driver** (e.g., `vault-java-driver-5.1.0.jar`) to the
+   `<APIM_HOME>/repository/components/lib/` directory.
+
+4. Create the `/repository/conf/security/secret-conf.properties` file and set the following configurations:
```

Comment:
```suggestion
4. Create the `/repository/conf/security/secret-conf.properties` file and set the following configurations.
```

Reply:
corrected in 1d8bdffa01d44e5da79703dd2060f0a621bb659a